src/valiant/metadata_utils.py

```diff
@@ -74,32 +107,36 @@ def get_slice(s: str, a: int, b: int) -> str:
 
 def slice_string_field(
     df: pd.DataFrame,
-    mask: pd.Series,
     string_col_name: str,
     start_col_name: str,
-    end_col_name: str
+    end_col_name: str,
+    mask: Optional[pd.Series] = None
 ) -> List[str]:
     return [
         get_slice(s, a, b)
         for s, (a, b) in zip(
-            df.loc[mask, string_col_name],
+            _get_field(df, string_col_name, mask=mask),
             zip(
-                df.loc[mask, start_col_name],
-                df.loc[mask, end_col_name]
+                _get_field(df, start_col_name, mask=mask),
+                _get_field(df, end_col_name, mask=mask)
```

Comment:
Could this be simplified down to the following?
```python
for s, a, b in zip(
    _get_field(df, string_col_name, mask=mask),
    _get_field(df, start_col_name, mask=mask),
    _get_field(df, end_col_name, mask=mask))
```

Reply:
It could and should, yes.